### ISSUE TITLE
Firebase Crash initialization fails

### ISSUE THREAD

**Issue by tymm** [scala-android/sbt-android-gms#1]
Im using:

- sbt-android 1.7.1
- sbt-android-gms 0.3
- google services 9.8.0
- firebase 9.8.0
- android support libraries 25.0.1

It compiles (and therefore finds `google-services.json`) and the app runs without any problems.
However FirebaseCrash.report(new Exception("Test")) will not show up in the Firebase Crash dashboard.
In the logs I see:
```
E/FirebaseCrash( 6308): Failed to initialize crash reporting: Missing an expected resource: 'R.string.google_api_key' for initializing Google services.  Possible causes are missing google-services.json or com.google.gms.google-services gradle plugin.
D/FirebaseCrash( 6308): Cannot register Firebase Analytics listener since Firebase Crash Reporting is not enabled
```

and
```
I/FirebaseCrashApiImpl( 6308): FirebaseCrashApiImpl created by ClassLoader com.google.android.chimera.container.internal.DelegateLastPathClassLoader[DexPathList[[zip file "/data/data/com.google.android.gms/app_chimera/m/0000000f/DynamiteModulesC_GmsCore_prodlmp_xhdpi_release.apk"],nativeLibraryDirectories=[/vendor/lib, /system/lib]]]
W/ModuleProvider( 1185): Failed to load module version: 
W/ModuleProvider( 1185): java.lang.IllegalArgumentException: module com.google.android.gms.flags not found
W/ModuleProvider( 1185): 	at com.google.android.chimera.container.ConfigurationManager.getModuleVersion(:com.google.android.gms:564)
W/ModuleProvider( 1185): 	at com.google.android.gms.chimera.container.ModuleProvider.query(:com.google.android.gms:1103)
W/ModuleProvider( 1185): 	at android.content.ContentProvider.query(ContentProvider.java:966)
W/ModuleProvider( 1185): 	at android.content.ContentProvider$Transport.query(ContentProvider.java:211)
W/ModuleProvider( 1185): 	at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:112)
W/ModuleProvider( 1185): 	at android.os.Binder.execTransact(Binder.java:446)
W/ModuleProvider( 1185): Failed to load module version after staging APKs
W/ModuleProvider( 1185): java.lang.IllegalArgumentException: module com.google.android.gms.flags not found
W/ModuleProvider( 1185): 	at com.google.android.chimera.container.ConfigurationManager.getModuleVersion(:com.google.android.gms:564)
W/ModuleProvider( 1185): 	at com.google.android.gms.chimera.container.ModuleProvider.query(:com.google.android.gms:1110)
W/ModuleProvider( 1185): 	at android.content.ContentProvider.query(ContentProvider.java:966)
W/ModuleProvider( 1185): 	at android.content.ContentProvider$Transport.query(ContentProvider.java:211)
W/ModuleProvider( 1185): 	at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:112)
W/ModuleProvider( 1185): 	at android.os.Binder.execTransact(Binder.java:446)
```

Manually supplying `R.string.google_api_key` results in successfully (?) loading Firebase Crash. However sending crash reports still fails.

```
I/FirebaseCrash( 7916): FirebaseCrash reporting initialized com.google.firebase.crash.internal.zze@28665a78
I/FirebaseInitProvider( 7916): FirebaseApp initialization successful
```

```
E/FirebaseCrash( 7916): Error sending crash report
E/FirebaseCrash( 7916): blj: Server did not receive report: Origin Error message: Invalid package version number.
E/FirebaseCrash( 7916): 	at blt.a(:com.google.android.gms.DynamiteModulesC:1053)
E/FirebaseCrash( 7916): 	at blz.a(:com.google.android.gms.DynamiteModulesC:5055)
E/FirebaseCrash( 7916): 	at blz.run(:com.google.android.gms.DynamiteModulesC:54)
E/FirebaseCrash( 7916): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/FirebaseCrash( 7916): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/FirebaseCrash( 7916): 	at java.lang.Thread.run(Thread.java:818)
```

**Comment by jorge2antibes** [comment]
Until this morning the app was running perfectly, just to find out this same issue.

I have updated the platform and repositories but still the same problem. It looks its related to firebase bug?!

**Comment by pfn** [comment]
Sounds like firebase updated the way it works and requires changes to the
plugin.

**Comment by pfn** [comment]
> E/FirebaseCrash( 7916): Error sending crash report
> E/FirebaseCrash( 7916): blj: Server did not receive report: Origin Error message: Invalid package version number.

This is because `versionCode` is not set properly (defaults to `-1` as I understand it, and that will cause this error).

missing `google_api_key` will be fixed in `0.4`
